### PR TITLE
MCR-3134 update dependencies

### PR DESCRIPTION
**Pull request by yagee-de**
[Link to jira](https://mycore.atlassian.net/browse/MCR-3134).

### CODE REVIEW

**Review by rsteph-de — DISMISSED** [review]
build successfully tested with my  own MyCoRe application

**Review by toKrause — COMMENTED** [review]
Issue tittle is a big lie.

This PR does much more than just "update dependencies". For example,

- it almost completely reverts MCR-2752 w/o justification (as there is no description of the issue)
- afait. it doesn't just update the HTTP dependencies, but introduces completely new abstractions

Should be broken up into multiple issues or at least be accompanied by a proper description of what's going on

File: mycore-base/src/main/java/org/mycore/services/http/MCRQueryParameter.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import java.util.stream.Stream;
+
+/**
+ * A record representing a query parameter.
+ *
+ * @see <a href="https://www.w3.org/TR/2014/REC-html5-20141028/forms.html#url-encoded-form-data">W3C recomandation</a>
+ * @param name query parameter name, null is allowed but transformed to empty String
+ * @param value query parameter name, null is allowed but transformed to empty String
+ */
+public record MCRQueryParameter(String name, String value) {
```
Comment (rsteph-de) [inline mycore-base/src/main/java/org/mycore/services/http/MCRQueryParameter.java]: Is this something we should implement ourselves?
I would prefer the use of an external API for that. For httpclient5 there is a hint to use UriBuilder.
https://stackoverflow.com/a/71477076

Comment (yagee-de) [reply mycore-base/src/main/java/org/mycore/services/http/MCRQueryParameter.java]: I would prefer not to mix dependencies. Here this class is used for JDK internal HttpClient. It's small and well tested. Coupling JDK internal HttpClient code with an external HttpClient library (here Apache HttpClient 5.x) would make it to hard to get rid of this dependency later.
IMHO the only advantage of Apache HttpClient at the moment is that's the only implementation thats supports caching. Neither Jetty HttpClient (required for SOLR) nor JDK internalt HttpClient support caching at the moment.

**Comment by yagee-de** [comment]
> Issue tittle is a big lie.
> 
> This PR does much more than just "update dependencies". For example,
> 
>     * it almost completely reverts MCR-2752 w/o justification (as there is no description of the issue)
> 
>     * afait. it doesn't just update the HTTP dependencies, but introduces completely new abstractions
> 
> 
> Should be broken up into multiple issues or at least be accompanied by a proper description of what's going on

mycore-bom is still there and works as expected. Please try it out. The current usage of mycore-bom was not OK, as it requires mycore-* modules to import mycore-bom within the same reactor built. This causes:


```
[INFO] Scanning for projects...
[WARNING]
[WARNING] Some problems were encountered while building the effective model for 'org.mycore:mycore-acl:jar:2024.06.0-SNAPSHOT'
[WARNING] BOM imports from within reactor should be avoided @ line 158, column 7
[WARNING]
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING]
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
```

This issue is fixed by using dependencyManagement in parent POM (mycore), still providing mycore-bom but not using it in other mycore-* modules.

I verified this behavior with a test project:
```
    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>org.mycore</groupId>
                <artifactId>mycore-bom</artifactId>
                <version>${mycore.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>

    <dependencies>
        <dependency>
            <groupId>org.mycore</groupId>
            <artifactId>mycore-base</artifactId>
        </dependency>
        <dependency>
            <groupId>org.apache.logging.log4j</groupId>
            <artifactId>log4j-api</artifactId>
        </dependency>
    </dependencies>
```

The log4j-api artifact gets resolved and does not need a version element.

**Comment by toKrause** [comment]
That might all be true? A few questions though:

- How is this a dependency update?
- How doesn't that revert MCR-2752 whose goal was: "die MyCoRe-Dependencies [sollen] in ein separates Maven-Modul `mycore-bom` herausgelöst werden"
- What is the point of still having `mycore-bom` when this PR has been applied? I appears to be unused and should consequently be removed (to be clear: this is not what I'm advocating for, but it would then be what would make sense to do)

**Comment by michael-becker** [comment]
Unfortunately, I cannot evaluate this PR since it contains way too much changes. It seems to update httpClient library dependency but also some other (log4j etc.). How does the new MCRQueryParameter record add to "update dependencies"?  We should totally split this PR in multiple PRs, for example
* update apache http client library
* introduce MCRQueryParameter implementation
* update BOM dependencies
* update mycore-pi module (it seems, package names were changed?)
* update MCRUserManager query

**Comment by yagee-de** [comment]
> That might all be true? A few questions though:
> 
>     * How is this a dependency update?

Update to dependencies sometime require code changes. Here: HttpClient 4.x is not compatible with HttpClient 5.x, It's not even required to use Apache HttpClient anymore, as there is HttpClient in Java 11 and higher, with one exception: Caching is only available in Apache HttpClient.

> 
>     * How doesn't that revert MCR-2752 whose goal was: "die MyCoRe-Dependencies [sollen] in ein separates Maven-Modul `mycore-bom` herausgelöst werden"
> 
>     * What is the point of still having `mycore-bom` when this PR has been applied? I appears to be unused and should consequently be removed (to be clear: this is not what I'm advocating for, but it would then be what would make sense to do)

Probably the mycore/pom.xml can be used as a valid replacement for mycore-bom. Don't know, not tested.

**Comment by toKrause** [comment]
> How is this a dependency update?

In that question (which was a direct reply to your previous post) "this" obviously was intended to refer to the complete overhaul of the approach to dependency management, not the update to HttpClient 5.x, as in: How is the complete overhaul of the approach to dependency management a dependency update?

**Comment by toKrause** [comment]
With [this additional change](https://github.com/MyCoRe-Org/mycore/pull/2219), the BOM would actually be a BOM and would be used as a BOM.

I've ran `mvn dependency:list -pl mycore-base` with and without that change, and the results are identical:

```
[INFO] The following files have been resolved:
[INFO]    com.auth0:java-jwt:jar:4.4.0:compile -- module com.auth0.jwt
[INFO]    com.bucket4j:bucket4j-core:jar:8.10.1:compile -- module io.github.bucket4j.core [auto]
[INFO]    com.fasterxml.jackson.core:jackson-annotations:jar:2.17.2:compile -- module com.fasterxml.jackson.annotation
[INFO]    com.fasterxml.jackson.core:jackson-databind:jar:2.17.2:compile -- module com.fasterxml.jackson.databind
[INFO]    com.fasterxml.jackson.core:jackson-core:jar:2.17.2:compile -- module com.fasterxml.jackson.core
[INFO]    com.fasterxml.woodstox:woodstox-core:jar:7.0.0:runtime -- module com.ctc.wstx
[INFO]    com.github.spotbugs:spotbugs-annotations:jar:4.8.6:provided -- module com.github.spotbugs.annotations [auto]
[INFO]    com.google.code.findbugs:jsr305:jar:3.0.2:compile -- module jsr305 (auto)
[INFO]    com.google.code.gson:gson:jar:2.11.0:compile -- module com.google.gson
[INFO]    com.google.errorprone:error_prone_annotations:jar:2.27.0:compile -- module com.google.errorprone.annotations
[INFO]    com.google.guava:guava:jar:33.2.1-jre:compile -- module com.google.common [auto]
[INFO]    com.google.guava:failureaccess:jar:1.0.2:compile -- module com.google.common.util.concurrent.internal [auto]
[INFO]    com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile -- module listenablefuture (auto)
[INFO]    org.checkerframework:checker-qual:jar:3.42.0:compile -- module org.checkerframework.checker.qual [auto]
[INFO]    com.google.j2objc:j2objc-annotations:jar:3.0.0:compile -- module com.google.j2objc.annotations
[INFO]    com.google.jimfs:jimfs:jar:1.3.0:test -- module com.google.common.jimfs [auto]
[INFO]    com.google.zxing:core:jar:3.5.3:compile -- module com.google.zxing [auto]
[INFO]    com.h2database:h2:jar:2.3.230:test -- module com.h2database [auto]
[INFO]    com.ibm.icu:icu4j:jar:75.1:compile -- module com.ibm.icu [auto]
[INFO]    com.jcraft:jsch:jar:0.1.55:runtime -- module jsch (auto)
[INFO]    commons-beanutils:commons-beanutils:jar:1.9.4:compile -- module commons.beanutils (auto)
[INFO]    commons-collections:commons-collections:jar:3.2.2:compile -- module commons.collections (auto)
[INFO]    commons-io:commons-io:jar:2.16.1:compile -- module org.apache.commons.io
[INFO]    io.github.classgraph:classgraph:jar:4.8.174:compile -- module io.github.classgraph
[INFO]    io.swagger.core.v3:swagger-annotations-jakarta:jar:2.2.22:compile -- module io.swagger.v3.oas.annotations [auto]
[INFO]    jakarta.activation:jakarta.activation-api:jar:2.1.3:compile -- module jakarta.activation
[INFO]    jakarta.annotation:jakarta.annotation-api:jar:3.0.0:compile -- module jakarta.annotation
[INFO]    jakarta.inject:jakarta.inject-api:jar:2.0.1:compile -- module jakarta.inject
[INFO]    jakarta.mail:jakarta.mail-api:jar:2.1.3:compile -- module jakarta.mail
[INFO]    jakarta.persistence:jakarta.persistence-api:jar:3.1.0:compile -- module jakarta.persistence
[INFO]    jakarta.servlet:jakarta.servlet-api:jar:6.0.0:provided -- module jakarta.servlet
[INFO]    jakarta.websocket:jakarta.websocket-api:jar:2.2.0:compile -- module jakarta.websocket
[INFO]    jakarta.websocket:jakarta.websocket-client-api:jar:2.2.0:compile -- module jakarta.websocket.client
[INFO]    jakarta.ws.rs:jakarta.ws.rs-api:jar:4.0.0:compile -- module jakarta.ws.rs
[INFO]    jakarta.xml.bind:jakarta.xml.bind-api:jar:4.0.2:compile -- module jakarta.xml.bind
[INFO]    jaxen:jaxen:jar:2.0.0:compile -- module org.jaxen [auto]
[INFO]    junit:junit:jar:4.13.2:test -- module junit [auto]
[INFO]    net.java.dev.jna:jna:jar:5.14.0:runtime -- module com.sun.jna [auto]
[INFO]    net.sf.saxon:Saxon-HE:jar:12.5:runtime -- module Saxon.HE (auto)
[INFO]    org.xmlresolver:xmlresolver:jar:5.2.2:runtime -- module org.xmlresolver.xmlresolver [auto]
[INFO]    org.xmlresolver:xmlresolver:jar:data:5.2.2:runtime -- module org.xmlresolver.xmlresolver_data [auto]
[INFO]    org.apache.commons:commons-compress:jar:1.26.2:compile -- module org.apache.commons.compress
[INFO]    commons-codec:commons-codec:jar:1.17.0:compile -- module org.apache.commons.codec
[INFO]    org.apache.commons:commons-lang3:jar:3.15.0:compile -- module org.apache.commons.lang3
[INFO]    org.apache.commons:commons-text:jar:1.12.0:compile -- module org.apache.commons.text
[INFO]    org.apache.httpcomponents.client5:httpclient5:jar:5.3.1:compile -- module org.apache.httpcomponents.client5.httpclient5 [auto]
[INFO]    org.apache.httpcomponents.core5:httpcore5-h2:jar:5.2.5:compile -- module org.apache.httpcomponents.core5.httpcore5.h2 [auto]
[INFO]    org.slf4j:slf4j-api:jar:1.7.36:compile -- module org.slf4j [auto]
[INFO]    org.apache.httpcomponents.client5:httpclient5-cache:jar:5.3.1:compile -- module org.apache.httpcomponents.client5.httpclient5.cache [auto]
[INFO]    org.apache.httpcomponents.core5:httpcore5:jar:5.2.5:compile -- module org.apache.httpcomponents.core5.httpcore5 [auto]
[INFO]    org.apache.logging.log4j:log4j-api:jar:2.23.1:compile -- module org.apache.logging.log4j
[INFO]    org.apache.logging.log4j:log4j-core:jar:2.23.1:compile -- module org.apache.logging.log4j.core
[INFO]    org.apache.logging.log4j:log4j-jakarta-web:jar:2.23.1:compile -- module org.apache.logging.log4j.web
[INFO]    org.codehaus.woodstox:stax2-api:jar:4.2.2:runtime -- module org.codehaus.stax2
[INFO]    org.eclipse.angus:angus-activation:jar:2.0.2:runtime -- module org.eclipse.angus.activation
[INFO]    org.eclipse.angus:angus-mail:jar:2.0.3:runtime -- module org.eclipse.angus.mail
[INFO]    org.glassfish.hk2:hk2-api:jar:3.1.1:compile -- module org.glassfish.hk2.api [auto]
[INFO]    org.glassfish.hk2:hk2-utils:jar:3.1.1:compile -- module org.glassfish.hk2.utilities [auto]
[INFO]    org.glassfish.jaxb:jaxb-runtime:jar:4.0.5:runtime -- module org.glassfish.jaxb.runtime
[INFO]    org.glassfish.jaxb:jaxb-core:jar:4.0.5:runtime -- module org.glassfish.jaxb.core
[INFO]    org.glassfish.jaxb:txw2:jar:4.0.5:runtime -- module com.sun.xml.txw2
[INFO]    com.sun.istack:istack-commons-runtime:jar:4.1.2:runtime -- module com.sun.istack.runtime
[INFO]    org.glassfish.jersey.core:jersey-server:jar:3.1.7:compile -- module jersey.server (auto)
[INFO]    org.glassfish.jersey.core:jersey-common:jar:3.1.7:compile -- module jersey.common (auto)
[INFO]    org.glassfish.hk2:osgi-resource-locator:jar:1.0.3:compile -- module osgi.resource.locator (auto)
[INFO]    jakarta.validation:jakarta.validation-api:jar:3.1.0:compile -- module jakarta.validation
[INFO]    org.glassfish.jersey.media:jersey-media-multipart:jar:3.1.7:compile -- module jersey.media.multipart (auto)
[INFO]    org.jvnet.mimepull:mimepull:jar:1.9.15:compile -- module org.jvnet.mimepull
[INFO]    org.hamcrest:hamcrest-core:jar:2.2:test -- module org.hamcrest.core.deprecated [auto]
[INFO]    org.hamcrest:hamcrest:jar:2.2:test -- module org.hamcrest [auto]
[INFO]    org.hibernate.orm:hibernate-core:jar:6.5.2.Final:compile -- module org.hibernate.orm.core [auto]
[INFO]    jakarta.transaction:jakarta.transaction-api:jar:2.0.1:compile -- module jakarta.transaction
[INFO]    org.jboss.logging:jboss-logging:jar:3.5.0.Final:runtime -- module org.jboss.logging
[INFO]    org.hibernate.common:hibernate-commons-annotations:jar:6.0.6.Final:runtime -- module org.hibernate.commons.annotations
[INFO]    io.smallrye:jandex:jar:3.1.2:runtime -- module org.jboss.jandex [auto]
[INFO]    com.fasterxml:classmate:jar:1.5.1:runtime -- module com.fasterxml.classmate
[INFO]    net.bytebuddy:byte-buddy:jar:1.14.15:runtime -- module net.bytebuddy
[INFO]    org.antlr:antlr4-runtime:jar:4.13.0:runtime -- module org.antlr.antlr4.runtime [auto]
[INFO]    org.javassist:javassist:jar:3.30.2-GA:runtime -- module org.javassist [auto]
[INFO]    org.jdom:jdom2:jar:2.0.6.1:compile -- module org.jdom2 [auto]
[INFO]    org.jsoup:jsoup:jar:1.18.1:compile -- module org.jsoup
[INFO]    org.mycore.xsonify:xsonify-serializer:jar:1.0:compile -- module org.mycore.xsonify.serialize
[INFO]    org.mycore.xsonify:xsonify-xml:jar:1.0:compile -- module org.mycore.xsonify.xml
[INFO]    org.mycore.xsonify:xsonify-xsd:jar:1.0:compile -- module org.mycore.xsonify.xsd
[INFO]    org.tmatesoft.svnkit:svnkit:jar:1.10.11:compile -- module svnkit (auto)
[INFO]    org.apache.sshd:sshd-core:jar:2.9.2:compile -- module sshd.core (auto)
[INFO]    org.slf4j:jcl-over-slf4j:jar:1.7.32:compile -- module org.apache.commons.logging [auto]
[INFO]    org.apache.sshd:sshd-common:jar:2.9.2:compile -- module sshd.common (auto)
[INFO]    net.i2p.crypto:eddsa:jar:0.3.0:compile -- module net.i2p.crypto.eddsa [auto]
[INFO]    de.regnis.q.sequence:sequence-library:jar:1.0.4:compile -- module sequence.library (auto)
[INFO]    org.tmatesoft.sqljet:sqljet:jar:1.1.15:compile -- module sqljet (auto)
[INFO]    org.antlr:antlr-runtime:jar:3.4:compile -- module antlr.runtime (auto)
[INFO]    net.java.dev.jna:jna-platform:jar:5.6.0:compile -- module com.sun.jna.platform [auto]
[INFO]    org.lz4:lz4-java:jar:1.4.1:compile -- module lz4.java (auto)
[INFO]    se.jiderhamn.classloader-leak-prevention:classloader-leak-prevention-core:jar:2.7.0:compile -- module classloader.leak.prevention.core (auto)
[INFO]    xalan:xalan:jar:2.7.3:compile
[INFO]    org.apache.logging.log4j:log4j-slf4j-impl:jar:2.23.1:runtime -- module org.apache.logging.log4j.slf4j.impl
[INFO]    org.glassfish.jersey.inject:jersey-hk2:jar:3.1.7:runtime -- module jersey.hk2 (auto)
[INFO]    org.glassfish.hk2:hk2-locator:jar:3.0.6:runtime -- module org.glassfish.hk2.locator [auto]
[INFO]    org.glassfish.hk2.external:aopalliance-repackaged:jar:3.1.1:provided -- module org.aopalliance [auto]
[INFO]    xalan:serializer:jar:2.7.3:runtime -- module serializer (auto)
[INFO]    com.zaxxer:HikariCP:jar:5.1.0:test -- module com.zaxxer.hikari
[INFO]    org.glassfish.grizzly:grizzly-http-server:jar:4.0.2:test -- module org.glassfish.grizzly.http.server
[INFO]    org.glassfish.grizzly:grizzly-http:jar:4.0.2:test -- module org.glassfish.grizzly.http
[INFO]    org.glassfish.grizzly:grizzly-framework:jar:4.0.2:test -- module org.glassfish.grizzly
[INFO]    org.glassfish.grizzly:grizzly-http-servlet:jar:4.0.2:test -- module org.glassfish.grizzly.servlet
[INFO]    org.glassfish.jersey.containers:jersey-container-grizzly2-http:jar:3.1.7:test -- module jersey.container.grizzly2.http (auto)
[INFO]    org.glassfish.jersey.containers:jersey-container-grizzly2-servlet:jar:3.1.7:test -- module jersey.container.grizzly2.servlet (auto)
[INFO]    org.glassfish.jersey.containers:jersey-container-servlet:jar:3.1.7:test -- module jersey.container.servlet (auto)
[INFO]    org.glassfish.jersey.containers:jersey-container-servlet-core:jar:3.1.7:test -- module jersey.container.servlet.core (auto)
[INFO]    org.glassfish.jersey.core:jersey-client:jar:3.1.7:test -- module jersey.client (auto)
[INFO]    org.glassfish.jersey.test-framework:jersey-test-framework-core:jar:3.1.7:test -- module jersey.test.framework.core (auto)
[INFO]    org.glassfish.jersey.media:jersey-media-jaxb:jar:3.1.7:test -- module jersey.media.jaxb (auto)
[INFO]    org.junit.jupiter:junit-jupiter:jar:5.10.3:test -- module org.junit.jupiter
[INFO]    org.junit.jupiter:junit-jupiter-params:jar:5.10.3:test -- module org.junit.jupiter.params
[INFO]    org.junit.jupiter:junit-jupiter-engine:jar:5.10.3:test -- module org.junit.jupiter.engine
[INFO]    org.glassfish.jersey.test-framework.providers:jersey-test-framework-provider-grizzly2:jar:3.1.7:test -- module jersey.test.framework.provider.grizzly2 (auto)
[INFO]    org.hibernate.orm:hibernate-hikaricp:jar:6.5.2.Final:test -- module org.hibernate.orm.hikaricp [auto]
[INFO]    org.junit.jupiter:junit-jupiter-api:jar:5.10.3:test -- module org.junit.jupiter.api
[INFO]    org.opentest4j:opentest4j:jar:1.3.0:test -- module org.opentest4j
[INFO]    org.junit.platform:junit-platform-commons:jar:1.10.3:test -- module org.junit.platform.commons
[INFO]    org.apiguardian:apiguardian-api:jar:1.1.2:test -- module org.apiguardian.api
[INFO]    org.junit.vintage:junit-vintage-engine:jar:5.10.3:test -- module org.junit.vintage.engine
[INFO]    org.junit.platform:junit-platform-engine:jar:1.10.3:test -- module org.junit.platform.engine
[INFO]    org.mycore:mycore-classifications:jar:1.1:test -- module mycore.classifications (auto)
[INFO]    org.postgresql:postgresql:jar:42.7.3:test -- module org.postgresql.jdbc [auto]
```

**Comment by toKrause** [comment]
Important for this to work: `org.mycore:mycore-bom` can not have `org.mycore:mycore` as a parent

File: mycore-base/src/test/java/org/mycore/services/http/MCRQueryParameterTest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+import org.junit.jupiter.api.Test;
+
+class MCRQueryParameterTest {
+    @Test
+    public void testOfEncodedString() {
+        MCRQueryParameter parameter = MCRQueryParameter.ofEncodedString("name=value");
+        assertEquals("name", parameter.name());
+        assertEquals("value", parameter.value());
+        parameter = MCRQueryParameter.ofEncodedString("name");
```
Comment (toKrause) [inline mycore-base/src/test/java/org/mycore/services/http/MCRQueryParameterTest.java]: additional test for edge case might be useful, probably:
```
parameter = MCRQueryParameter.ofEncodedString("name=");
assertEquals("name", parameter.name());
assertEquals("", parameter.value());
```

File: mycore-base/src/main/java/org/mycore/services/http/MCRQueryParameter.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+     * @return The encoded string, empty if parameters are empty
+     * @throws NullPointerException if parameters are empty or contains null values
+     */
+    public static String toQueryString(List<MCRQueryParameter> parameters) {
+        if (parameters.isEmpty()) {
+            return "";
+        }
+        return parameters.stream()
+            .map(MCRQueryParameter::toEncodedString)
+            .collect(Collectors.joining("&", "?", ""));
```
Comment (rsteph-de) [inline mycore-base/src/main/java/org/mycore/services/http/MCRQueryParameter.java]: query should not start with `?`

java.net.URI also does not contain `?` in query string
``` java
URI uri = new URI("http://www.test.de?a=1&b=2");
System.out.println(uri.getQuery()); -> "a=1&b=2"
````

File: mycore-pi/src/main/java/org/mycore/pi/doi/client/crossref/MCRCrossrefClient.java [hunk 4]
```diff
@@ -119,41 +121,49 @@ public void doMDUpload(Document metadata) throws MCRPersistentIdentifierExceptio
         postRequest.setEntity(reqEntity);
 
         try (CloseableHttpClient client = getHttpClient()) {
-            try (CloseableHttpResponse response = client.execute(postRequest)) {
-                final int statusCode = response.getStatusLine().getStatusCode();
+            client.<MCRResultOrException<Object, MCRPersistentIdentifierException>>execute(postRequest, response -> {
```
Comment (toKrause) [inline mycore-pi/src/main/java/org/mycore/pi/doi/client/crossref/MCRCrossrefClient.java]: should be `Void` insted of `Object`

File: mycore-pi/src/main/java/org/mycore/pi/doi/client/datacite/MCRDataciteClient.java [hunk 5]
```diff
@@ -136,34 +139,40 @@ public List<Map.Entry<String, URI>> getMediaList(final MCRDigitalObjectIdentifie
...
-            switch (statusLine.getStatusCode()) {
-                case HttpStatus.SC_OK -> {
-                    try (Scanner scanner = new Scanner(response.getEntity().getContent(), StandardCharsets.UTF_8)) {
-                        while (scanner.hasNextLine()) {
-                            String line = scanner.nextLine();
-                            String[] parts = line.split("=", 2);
-                            String mediaType = parts[0];
-                            URI mediaURI = new URI(parts[1]);
-                            entries.add(new AbstractMap.SimpleEntry<>(mediaType, mediaURI));
+            MCRResultOrException<List<Map.Entry<String, URI>>, MCRPersistentIdentifierException> result
```
Comment (toKrause) [inline mycore-pi/src/main/java/org/mycore/pi/doi/client/datacite/MCRDataciteClient.java:142]: is this correct? `List<Map.Entry<String, URI>`, could `Map<String,List<String>>`, ...,  but doesn't have to be fixed in this PR

**Review by toKrause — CHANGES_REQUESTED** [review]

File: mycore-base/src/main/java/org/mycore/services/http/MCRQueryParameter.java (hunk 1 above)
Comment (rsteph-de) [reply mycore-base/src/main/java/org/mycore/services/http/MCRQueryParameter.java]: Can we rename this to `MCRURLQueryParameter`?
The term *MCRQuery* means something different in the MyCoRe world

File: mycore-pi/src/main/java/org/mycore/pi/doi/client/crossref/MCRCrossrefClient.java (hunk 4 above)
Comment (yagee-de) [reply mycore-pi/src/main/java/org/mycore/pi/doi/client/crossref/MCRCrossrefClient.java]: fixed in https://github.com/MyCoRe-Org/mycore/pull/2212/commits/669e4f6d271ff9a5718a9c3ae8db413811af7266

File: mycore-base/src/main/java/org/mycore/services/http/MCRQueryParameter.java (hunk 3 above)
Comment (yagee-de) [reply mycore-base/src/main/java/org/mycore/services/http/MCRQueryParameter.java]: fixed in https://github.com/MyCoRe-Org/mycore/pull/2212/commits/669e4f6d271ff9a5718a9c3ae8db413811af7266

File: mycore-base/src/test/java/org/mycore/services/http/MCRQueryParameterTest.java (hunk 2 above)
Comment (yagee-de) [reply mycore-base/src/test/java/org/mycore/services/http/MCRQueryParameterTest.java]: added test for parsing "name=" in https://github.com/MyCoRe-Org/mycore/pull/2212/commits/669e4f6d271ff9a5718a9c3ae8db413811af7266

**Review by toKrause — DISMISSED** [review]

**Comment by sebhofmann** [comment]
A relevant change:
Relevant Change from solr 9.0:
The response format for field values serialized as raw XML (via the [xml] raw value DocTransformer and wt=xml) has changed. Previously, values were dropped in directly as top-level child elements of each <doc>, obscuring associated field names and yielding inconsistent <doc> structure. As of version 9.0, raw values are wrapped in a <raw name="field_name">[…​]</raw> element at the top level of each <doc> (or within an enclosing <arr name="field_name"><raw>[…​]</raw></arr> element for multi-valued fields). Existing clients that parse field values serialized in this way will need to be updated accordingly.

**Review by toKrause — APPROVED** [review]

**Comment by yagee-de** [comment]
> A relevant change: Relevant Change from solr 9.0: The response format for field values serialized as raw XML (via the [xml] raw value DocTransformer and wt=xml) has changed. Previously, values were dropped in directly as top-level child elements of each , obscuring associated field names and yielding inconsistent structure. As of version 9.0, raw values are wrapped in a […​] element at the top level of each (or within an enclosing […​] element for multi-valued fields). Existing clients that parse field values serialized in this way will need to be updated accordingly.

I don't think that we serialize values as "raw XML". I don't notice a problem upgrading to 9.6 on client and server side so far

**Review by rsteph-de — APPROVED** [review]
